src/decoder/lossless.rs

```diff
@@ -0,0 +1,212 @@
+use decoder::{Decoder, MAX_COMPONENTS};
+use error::{Error, Result, UnsupportedFeature};
+use huffman::{fill_default_mjpeg_tables, HuffmanDecoder, HuffmanTable};
+use marker::Marker;
+use parser::Predictor;
+use parser::{
+    parse_app, parse_com, parse_dht, parse_dqt, parse_dri, parse_sof, parse_sos,
+    AdobeColorTransform, AppData, CodingProcess, Component, Dimensions, EntropyCoding, FrameInfo,
+    IccChunk, ScanInfo,
+};
+use std::io::Read;
+
+impl<R: Read> Decoder<R> {
+    /// decode_scan_lossless
+    pub fn decode_scan_lossless(
+        &mut self,
+        frame: &FrameInfo,
+        scan: &ScanInfo,
+    ) -> Result<(Option<Marker>, Option<Vec<Vec<u16>>>)> {
```

Comment:
I can't see any return path making use of `Option<Vec<_>>` here. In any case, when the types become this complicated I found it better to create a very simple struct and be it just for the ability to properly document each portion of the return type as field documentation.

Reply:
It's the same return type the `decode_scan` function in `decoder.rs` has. However I now removed the Option around `Vec<Vec<u16>>`.

Reply:
Gotcha, I see why it was initially chosen. Although, this increases the benefit of having a new type defined for this instance since it could be shared between methods.